src/HttpOverStream.NamedPipe/NamedPipeListener.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using System.IO;
+using System.IO.Pipes;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace HttpOverStream.NamedPipe
+{
+    public class NamedPipeListener : IListen
+    {
+        private Task _listenTask;
+        private CancellationTokenSource _listenTcs;
+        private readonly string _pipeName;
+        private readonly PipeDirection _pipeDirection;
+        private readonly PipeOptions _pipeOptions;
+        private readonly PipeTransmissionMode _pipeTransmissionMode;
+        private readonly int _maxAllowedServerInstances;
+
+        public NamedPipeListener(string pipeName)
+            : this(pipeName, PipeDirection.InOut, PipeOptions.Asynchronous | PipeOptions.WriteThrough, PipeTransmissionMode.Byte, 0)
+        {
+        }
+        public NamedPipeListener(string pipeName, PipeDirection pipeDirection, PipeOptions pipeOptions, PipeTransmissionMode pipeTransmissionMode, int maxAllowedServerInstances)
+        {
+            _pipeName = pipeName;
+            _pipeDirection = pipeDirection;
+            _pipeOptions = pipeOptions;
+            _pipeTransmissionMode = pipeTransmissionMode;
+            _maxAllowedServerInstances = maxAllowedServerInstances;
+        }
+
+        public Task StartAsync(Action<Stream> onConnection, CancellationToken cancellationToken)
+        {
+            // TODO: use the parameter cancellationToken
```

Comment:
This was copied across from our codebase (i didnt add the TODO comment).

Reply:
Actually the TODO is not usefull anymore (we already handle cancellation)

Reply:
To clarify, if someone cancels the token passed into StartAsync, it won't do anything here. Is that OK?